java/gateway-core/src/main/java/com/avaulta/gateway/pseudonyms/impl/Sha256PseudonymEncoder.java

```diff
@@ -0,0 +1,43 @@
+package com.avaulta.gateway.pseudonyms.impl;
+
+import com.avaulta.gateway.pseudonyms.Pseudonym;
+import com.avaulta.gateway.pseudonyms.PseudonymEncoder;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * implementation of defacto encoding used by BulkDataSanitizerImpl as of v0.4.30
+ */
+public class Sha256PseudonymEncoder implements PseudonymEncoder {
+
+
+    @Override
+    public String encode(Pseudonym pseudonym) {
+        return base64Encode(pseudonym.getHash());
+    }
+
+    @Override
+    public Pseudonym decode(String input) {
+        return Pseudonym.builder()
+            .hash(input.getBytes(StandardCharsets.UTF_8))
```

Comment:
doesn't this need to be decoded??
```suggestion
            .hash(base64decoder.decode(input.getBytes(StandardCharsets.UTF_8)))
```

Reply:
just made up the `base64decoder` part